client/my-sites/checklist/checklist-header.jsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import ScreenReaderText from 'components/screen-reader-text';
+import ProgressBar from 'components/progress-bar';
+
+let globalIndex = 0;
+
+export class ChecklistHeader extends PureComponent {
+	static propTypes = {
+		total: PropTypes.number.isRequired,
+		completed: PropTypes.number.isRequired,
+		hideCompleted: PropTypes.bool,
+		onClick: PropTypes.func,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.index = globalIndex++;
+	}
+
+	render() {
+		const { completed, hideCompleted, total, translate } = this.props;
+		const buttonText = hideCompleted
+			? translate( 'Show completed' )
+			: translate( 'Hide completed' );
+
+		/* eslint-disable wpcalypso/jsx-classname-namespace */
```

Comment:
Don't forget to remove these 😄

Reply:
I tried it. :)
But removing that line caused a bunch of lint errors that said the class names should start with "checklist__".

Reply:
That's unexpected, would it be possible to double check that the display name for the component is `checklist-header`? Maybe I'm mis-reading something here @aduth?

Reply:
It is expected. Only the default export of a folder's `index.js(x)` ([or `main.js(x)`](https://github.com/Automattic/wp-calypso/blob/01c47104a3524be0657fdcfe0d33b319c17e4758/.eslintrc.js#L28-L30)) can have a `className` without `__` separator. All other components, including the root return value of a non-index file _must_ use the separator.

Reply:
Ah missed that it wasn't `checklist__header` here 😓

Reply:
Let's see this might involve a number of renames here then. `checklist__header`, `checklist__header-main`, `checklist__header-progress` etc.